### PR TITLE
feat(TDI-47398): migrate adal4j to msal4j mscrm

### PR DESCRIPTION
**Pull request by dmytro-sylaiev**
**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/pom.xml [hunk 1]
```diff
@@ -164,6 +159,19 @@
             <artifactId>jackson-annotations</artifactId>
             <version>${jackson.version}</version>
         </dependency>
+		<!-- https://mvnrepository.com/artifact/com.microsoft.azure/msal4j -->
```
Comment (ozhelezniak-talend) [inline main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/pom.xml:162]: what's wrong with spaces here?

File: main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/src/main/java/com/microsoft/aad/msal4j/OauthClientApplication.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    private ClientAuthentication clientAuthentication;
+    private CustomJWTAuthentication customJWTAuthentication;
+    private boolean clientCertAuthentication = false;
+    private ClientCertificate clientCertificate;
+    private boolean sendX5c;
+    private final String username;
+    private final String password;
+
+
+    @Override public CompletableFuture<IAuthenticationResult> acquireToken(ClientCredentialParameters parameters) {
```
Comment (ozhelezniak-talend) [inline main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/src/main/java/com/microsoft/aad/msal4j/OauthClientApplication.java:31]: Could you make it modifier list homogeneous in one file?
Here annotation is on the same line with modifiers but in the "clientAuthentication" annotation is on a separate line.
Also, it looks a bit weird that those two methods are before the constructor.

File: main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/src/main/java/com/microsoft/aad/msal4j/OauthClientApplication.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        validateNotNull("parameters", parameters);
+
+        RequestContext context = new RequestContext(this, PublicApi.ACQUIRE_TOKEN_FOR_CLIENT, parameters);
+        OauthCredentialRequest clientCredentialRequest = new OauthCredentialRequest(parameters,
+                username, password,this, context);
+        new OAuthAuthorizationGrant(null, null);
```
Comment (ozhelezniak-talend) [inline main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/src/main/java/com/microsoft/aad/msal4j/OauthClientApplication.java:37]: Is it some kind of check?
you create a new instance but don't use it.

**Review by dmytrogrygorenko — APPROVED** [review]
LGTM

**Pull request merged**